src/Features/Core/SolutionCrawler/WorkCoordinator.AsyncDocumentWorkItemQueue.cs

```diff
@@ -104,6 +95,33 @@ private DocumentId GetBestDocumentId_NoLock(ProjectId preferableProjectId)
                     return lowPriorityDocumentId;
                 }
 
+                private ProjectId GetBestProjectId_NoLock(ProjectId projectId, ProjectDependencyGraph dependencyGraph)
+                {
+                    if (projectId != null)
+                    {
+                        if (_documentWorkQueue.ContainsKey(projectId))
+                        {
+                            return projectId;
+                        }
+
+                        foreach (var dependingProjectId in dependencyGraph.GetProjectsThatDirectlyDependOnThisProject(projectId))
```

Comment:
Basically we are preferring dependent projects if no pending work for active project, right?